src/attr/_funcs.py

```diff
@@ -366,17 +366,22 @@ def resolve_types(cls, globalns=None, localns=None):
     :param type cls: Class to resolve.
     :param Optional[dict] globalns: Dictionary containing global variables.
     :param Optional[dict] localns: Dictionary containing local variables.
+    :param Optional[list] attribs: List of attribs for the given class.
+        This is necessary when calling from inside a ``field_transformer``
+        since *cls* is not an ``attrs`` class yet.
 
     :raise TypeError: If *cls* is not a class.
     :raise attr.exceptions.NotAnAttrsClassError: If *cls* is not an ``attrs``
-        class.
+        class and you didn't pass any attribs.
     :raise NameError: If types cannot be resolved because of missing variables.
 
     :returns: *cls* so you can use this function also as a class decorator.
         Please note that you have to apply it **after** `attr.s`. That means
         the decorator has to come in the line **before** `attr.s`.
 
     ..  versionadded:: 20.1.0
+    ..  versionadded:: 20.4.0 *attribs*
```

Comment:
```suggestion
    ..  versionadded:: 21.1.0 *attribs*
```

I know. It's still March 2020.